infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -266,13 +257,37 @@ private void createSessions() {
                 incomingSessions.size(), sessions);
     }
 
+
     private void logNewlyAddedSessionInfo(Set<LogReplicationSession> newlyAddedSessions) {
         log.info("========= HashCode -> Session mapping for newly added session: =========");
         for (LogReplicationSession session : newlyAddedSessions) {
             log.info("HashCode: {}, Session: {}", session.hashCode(), session);
         }
     }
 
+    /**
+     * Build a session and add to incoming/outgoing session appropriately.
+     */
+    private void constructSession(String sourceClusterId, String sinkClusterId,
+                                  Set<LogReplicationSession> sessionsToAdd, TxnContext txn,
+                                  Map<String, ClusterDescriptor> connectionEndpoints, boolean isIncoming) {
+        LogReplicationSession session = LogReplicationSession.newBuilder()
+                .setSourceClusterId(sourceClusterId)
+                .setSinkClusterId(sinkClusterId)
+                .setSubscriber(getDefaultSubscriber())
+                .build();
```

Comment:
Seems like `sessionsToAdd` is the newly constructed sessions, would it be more clear if we return it instead of passing a reference?

Reply:
this bit was removed to avoid confusion about outgoing sessions.